lib/introspection.js

```diff
@@ -1,91 +1,158 @@
+  for (let i = lines.length - 2; i >= 0; i--) {
+    const line = lines[i].trimLeft();
+    if (!line.startsWith('//')) break;
+    comments.unshift(line.slice(3));
+  }
+  return comments.map(line => line.trimRight()).filter(line => line.length);
+};
+
+const isNamedLine = line => NAMED_LINES.find(s => line.startsWith(s));
+
+// Parse comments related to function
+//   lines - <string[]>
+// Returns: <object>, function signature
+const parseComments = lines => {
+  const sig = {
+    title: lines[0] || '',
+    description: '',
+    parameters: [],
+    comments: [],
+  };
+  let i;
+  let paramStart = lines.length;
+  let paramEnd = lines.length;
+  let paramsStarted = false;
+
+  for (i = 1; i < lines.length; i++) {
+    if (!paramsStarted) {
+      if (lines[i].startsWith(' ')) {
+        paramStart = i;
+        paramsStarted = true;
+      } else if (isNamedLine(lines[i])) {
+        paramStart = i;
+        paramEnd = i;
+        break;
+      }
+    } else if (paramsStarted && !lines[i].startsWith(' ')) {
+      paramEnd = i;
+      break;
+    }
+  }
 
-const parseLines = (
-  // Parse signature lines
-  s, // string, signature lines
-  signature // record, { title, description, parameters, comments }
-  // Returns: array of string
-) => {
-  let lines = s.split('\n');
-  lines.pop();
-  signature.title = (lines.shift() || '').replace('//', '').trim();
-  lines = lines.map(
-    d => d.trim().replace(/^(.*) \/\//, '$1:').replace(',:', ':')
-  );
-  for (let line of lines) {
-    if (line.startsWith('//')) {
-      line = line.replace(/^\/\/ /, '').trim();
-      if (NAMED_LINES.find(s => line.startsWith(s))) {
-        const [name, comment] = common.section(line, ': ');
-        signature.comments.push({ name, comment });
-      } else if (signature.parameters.length === 0) {
-        if (signature.description.length > 0) {
-          signature.description += '\n';
+  sig.description = lines.slice(1, paramStart).join('\n');
+
+  for (i = paramStart; i < paramEnd; i++) {
+    const len = lines[i].length;
+    const arg = lines[i].trimLeft();
+    const offset = len - arg.length;
+    const [name, rest] = common.section(arg, ' - ');
+
+    if (rest !== '') {
+      let [type, comment] = common.section(rest, ', ');
+      const types = type.split(' | ').map(s => s.slice(1, s.length - 1));
+      if (types.length === 1) {
+        type = types[0];
+        if (!(ALL_TYPES.has(type) ||
+            (ALL_TYPES.has(type.slice(0, type.length - 2)) &&
+              type.slice(type.length - 2) === '[]'))) {
+          comment = rest;
+          type = '';
         }
-        signature.description += line;
       } else {
-        const par = last(signature.parameters);
-        par.comment += '\n' + line;
+        type = [];
+        const notValidTypes = [];
+        types.forEach(s => {
+          if (ALL_TYPES.has(s) ||
+            (ALL_TYPES.has(s.slice(0, s.length - 2) &&
+              s.slice(s.length - 2) === '[]'))) {
```

Comment:
```js
const arrType = type.slice(0, type.length - 2)
if (ALL_TYPES.has(type) ||
   (ALL_TYPES.has(arrType) && type.slice(type.length - 2) !== '[]')) {
```
(also parens around `(ALL_TYPES.has(arrType) && type.slice(type.length - 2) !== '[]')` can be omitted if you want)